File: Python/Class_tasks/Kata.py



def factorOf(number):
	sum = 0
	count = 1
	while(count <= number):
		if( number % count == 0):
			sum =+ 1
		count =+ 1


	return sum

		
				



def isPrime(number):
	count = 2
	sum = 0
	while( count < number):
		if (number % count == 0):
			sum =+ 1
		count =+ 1
			
	if (sum > 0):
		return false
	else:
		return true
			
			
			




def isOdd(number):

	if( number % 2 != 0):
		return true
	else:
		return false




def isLeapYear(year):
	check = 0
	if((year % 4 == 0) and (year % 100 != 0)):
		check = 1
	if( (year % 4 == 0) and (year % 100 == 0)):
		check = 2
	if (truef == 2 and year % 400 == 0):
		return true

	else:
		return false



def toConvert(fahrenheit):

	celcius = (5 * (fahrenheit - 32)) / 9
	return celcius




def printStars(number_of_rows):
	
	check = number_of_rows - (number_of_rows - 1)
	while(number_of_rows >= 0):
		count = 1
		while( count <= check):
			print("*")
			count =+ 1
		print("/n")
		check =+ 1
	
		number_of_rows -= 1


def isEven(number):

	if( number % 2 == 0):
		return true
	else:
		return false

		 

def subtract(number_1, number_2):
	max = 0
	minimum = 0
	equal = false
	if( number_1 > number_2):
		max = number_1
		minimum = number_2
	else:
		max = number2
		minimum = number1
	

	difference = max - minimum
	return(difference)
				


	
	

#def isSquare(number):
#	square = 0
#	notSquare = 0
#	for(count = 0; count < number; count++)
		
#		if( (count * count) == number):
#			square = 1
#			break	
		
#	if(square == 1):
#		return true
#	else
#		return false
	




def isPalindrome(number):	
			
	first_2_Numbers = number / 1000
	last_2_Numbers = number % 100
	first_Digit = first_2_Numbers / 10
	second_Digit = first_2_Numbers % 10
	fourth_Digit = last_2_Numbers / 10
	fifth_Digit = last_2_Numbers % 10


	if(first_Digit == fifth_Digit and second_Digit == fourth_Digit):
		return true
	else:
		return false




def factorialOf(number):
	factorial = number
	while (number < 1):
		factorial = factorial * (number - 1)
		number -= 1
	return factorial
	


def divide(number_1, number_2):

	qoutient = number_1 / number_2

	qoutient = number_1 / number_2

	if( number_2 == 0):
		qoutient = 0
	return qoutient






def squareOf(number):
	square = number * number
	return(square)



def get_number_of_boxes(num_of_people, number_of_slices):
	number_of_boxes = num_of_people / number_of_slices
	if (num_of_people % number_of_slices != 0) :
		number_of_boxes += 1
	return (number_of_boxes)


#def (num_of_people, number_of_slices):
#	slicesServed = number_of_slices * num_of_people
#	return (slicesServed)




def get_money_spent_on_pizza(number_of_boxes, price_per_box):
	bill = number_of_boxes * price_per_box
	return (bill)



def get_left_over_pizza(num_of_people, number_of_slices, number_of_boxes):
	leftover = (number_of_slices * number_of_boxes) - num_of_people
	return (leftover)

File: java/Class_tasks/Kata.java
public class Kata{


public static int factorOf(int number){
	int sum = 0;
	int count = 1;
	while(count <= number){
		if( number % count == 0)
			sum++;
		count++;}

	return sum;
				
}


public static boolean isPrime(int number){
	int count = 2;
	int sum = 0;
	while( count < number){
		if (number % count == 0)
			sum++;
		count++;}
			
	if (sum > 0)
		return false;
	else
		return true;
			
			
}

public static boolean isOdd(int number){

	if( number % 2 != 0)
		return true;
	else
		return false;

}


/* public static boolean isLeapYear(int year){
	int check = 0;
	if((year % 4 == 0) && year % 100 != 0)
		check = 1;
	if( (year % 4 == 0) && year % 100 == 0)
		check = 2;
	if (truef == 2 && year % 400 == 0)
		return true;

	else
		return false;

}
*/
public static double toConvert(double fahrenheit){

	double celcius = (5 * (fahrenheit - 32)) / 9;
	return celcius;


}

/* public static void printStars(int numberOfRows){
	
	int check = numberOfRows - (numberOfRows - 1);
	while(numberOfRows >= 0)
		int count = 1;
		while( count <= check){
			System.out.print("*");
			count++;}
		System.out.println();
		check++;
	
		numberOfRows--;

}

*/
public static boolean isEven(int number){

	if( number % 2 == 0)
		return true;
	else
		return false;

}	 

public static int subtract(int number1, int number2){
	int max = 0;
	int minimum = 0;

	if( number1 > number2){
		max = number1;
		minimum = number2;}
	else{
		max = number2;
		minimum = number1;}
	

	int difference = max - minimum;
	return(difference);
				

}

	

public static boolean isSquare(int number){
	int square = 0;
	int notSquare = 0;
	for(int count = 0; count < number; count++)
		
		if( (count * count) == number){
			square = 1;
			break;	}
		
	if(square == 1)
		return true;
	else
		return false;
	

}


public static boolean isPalindrome(int number){	
			
	int first2Numbers = number / 1000;
	int last2Numbers = number % 100;
	int firstDigit = first2Numbers / 10;
	int secondDigit = first2Numbers % 10;
	int fourthDigit = last2Numbers / 10;
	int fifthDigit = last2Numbers % 10;


	if(firstDigit == fifthDigit && secondDigit == fourthDigit)
		return true;
	else
		return false;

}


public static long factorialOf(int number){
	int factorial = number;
	while (number < 1){
		factorial = factorial * (number - 1);
		number--;}
	return factorial;
}
	


/* public static float divide(number1, number2)
	float qoutient = number1 / number2;
	if( number2 == 0)
		qoutient = 0;
	return quotient;
*/





public static long squareOf(int number){
	int square = number * number;
	return(square);
}

public static int getNumberOfBoxes(int numOfPeople, int numberOfSlices){
	int numberOfBoxes = numOfPeople / numberOfSlices;
	if (numOfPeople % numberOfSlices != 0) 
		numberOfBoxes++;
	return (numberOfBoxes);
}

public static int getNumberOfSlicesServed(int numberOfSlices, int numOfPeople){
	int slicesServed = numberOfSlices * numOfPeople;
	return slicesServed;

}


public static int getMoneySpentOnPizza(int numberOfBoxes, int pricePerBox){
	int bill = numberOfBoxes * pricePerBox;
	return bill;


}

public static int getLeftOverPizza(int numOfPeople, int numberOfSlices, int numberOfBoxes){
	int leftover = (numberOfSlices * numberOfBoxes) - numOfPeople;
	return (leftover);
}

public static int findSmallest(int array[]){
	int lastIndex = array.length - 1;
	int smallestInt = array[0];

	while( lastIndex >= 0){
		if( array[lastIndex] < smallestInt)
			smallestInt = array[lastIndex];
		lastIndex--;}
	return (smallestInt);	
}

public static double averageArray(int array[]){
	int lastIndex = array.length - 1;
	int sum = 0;
	while( lastIndex >= 0 ){
		sum += array[lastIndex];
		lastIndex++;}
	double average = sum / (array.length);
	
}

public static int countOccurence(int array[], int targetNumber){
	int occurence = 0;
	for(int index = 0; index < array.length; index++){
		if( targetNumber == array[index] )
			occurence++;
		} 

	return (occurence);


}


public static boolean containsElement(int array[], int targetNumber){
	for(int index = 0; index < array.length; index++){
		if( targetNumber == array[index] )
			return (true);
		else
			return (false); }


}


public static int getFirstElement(int array[]){
	int firstElement = array[0];
	if( array == array[0])
		return (0);
	else
		return (firstElement);


}


public static int getLastElement(int array[]){
	lastElement = array[(array.length - 1)];
	if( array == array[0])
		return (0);
	else
		return (lastElement);


}


public static int arrayLength(int array[]){
	int arrayLength = 0;

	for(int index = 0; index < array.length; index++){
		arrayLength++;
			}
	return(arrayLength);
		

}

public static int getMiddleElement(int array[]){
	int arrayLength = array.length;
	int middleIndex = arrayLength / 2;

	if( arrayLength % 2 == 0)
		return(new int []{array[middleIndex], array[middleIndex + 1]});
	else
		return(array[middleIndex]);
	
	
	}

public static int swapFirstAndLast(int array[]){
	int firstNumber = array[0];
	int lastNumber = array[array.length - 1];

	array[array.length - 1] = firstNumber;
	array[0] = lastNumber;

	return(array);


}

/* public static int findNumberOfIndex(int[]){
	int count = 0;
	while( int
	*/





}

